### PR TITLE
Serving an error page on 404.

### PR DESCRIPTION
**Pull request by JustusAdam**
Created a very rudimentary 404 page that gets served if a file is not found.

### CODE REVIEW

**Comment by vilterp** [comment]
Thanks for working on this! Could you upload a screenshot for convenience? Thanks.

**Comment by JustusAdam** [comment]
screen shot 2015-07-28 at 10 12 06

File: backend/Main.hs [hunk 1]
```diff
@@ -109,8 +110,9 @@ error400 =
 
 
 error404 :: Snap ()
-error404 =
+error404 = do
     modifyResponse $ setResponseStatus 404 "Not Found"
+    serveHtml $ Compile.htmlDocument "File not found" $ H.div $ H.h1 (H.toHtml ("File not found" :: String)) <> H.p (H.toHtml ("The file you requested could not be found" :: String))
```
Comment (vilterp) [inline backend/Main.hs]: Can you add line breaks to this to end it under 80 chars?

**Comment by JustusAdam** [comment]
Oh yeah sure.

**Comment by vilterp** [comment]
Thanks. Also it would be cool if the styling looked like the index page you made. After that I think we can merge this in!

**Comment by JustusAdam** [comment]
Hmm. I am not sure that is as trivial as it may seem. This is just a simple static html made with blaze, whereas the index uses elm only. Because of the choices  we made with the styling (inline) there's no css that I could just apply to the 404. 

I'd have to redo the 404 page and instead serve the index with some special payload that makes it display the error message. 
Theoretically possible and probably cool but that'll take some more time.

Would you like to add this as a temporary solution, just so the user gets at least some feedback and I'll come up with a cleaner, more beautiful solution a little later?

**Comment by evancz** [comment]
Alright, added a 404 page. Thanks again for working on this stuff!

**Pull request closed without merging**